spinalcordtoolbox/moco.py

```diff
@@ -853,3 +856,128 @@ def spline(folder_mat, nt, nz, verbose, index_b0 = [], graph=0):
+
+    # Build absolute output path and go to tmp folder
+    curdir = os.getcwd()
+    path_out_abs = os.path.abspath(param.path_out)
+    os.chdir(path_tmp)
+
+    # Load data and get its dimensions
+    sct.printv('\nGet dimensions of data...', param.verbose)
+    im_data = Image(file_data)
+    nx, ny, nz, nt, px, py, pz, pt = im_data.dim
+    sct.printv('  ' + str(nx) + ' x ' + str(ny) + ' x ' + str(nz) + ' x ' + str(nt), param.verbose)
+
+    # Split 4D data along z dimension
+    #sct.printv('\nSplit along z dimension...', param.verbose)
+    im_data_split_list = split_data(im_data, 2)  # 0 - x, 1 - y, 2 - z, 3 - t
+
+    sct.printv('\nWARNING: Data were acquired in interleaved mode.', param.verbose, 'warning')
+    sct.printv('Split data into two datasets (even and odd slices) and run moco separately...', param.verbose)
+
+    # Get only even slices across all volumes
+    data_even = []
+    # loop across even slices
+    for index_even in range(0, len(im_data_split_list), 2):
+        data_even.append(im_data_split_list[index_even])
+    concat_data(data_even, dim=2).save(file_data_even, verbose=0)  # Concatenate in z-axis and save
+
+    # Get only odd slices across all volumes
+    data_odd = []
+    # loop across odd slices
+    for index_odd in range(1, len(im_data_split_list), 2):
+        data_odd.append(im_data_split_list[index_odd])
+    concat_data(data_odd, dim=2).save(file_data_odd, verbose=0)  # Concatenate in z-axis and save
+
+    # Split mask if was passed
+    if param.fname_mask != '':
+        mask_data = Image(file_mask)
+        # Split 4D data along z dimension
+        im_mask_split_list = split_data(mask_data, 2)  # 0 - x, 1 - y, 2 - z, 3 - t
+
+        mask_even = []
+        # loop across even slices
+        for index_even in range(0, len(im_mask_split_list), 2):
+            mask_even.append(im_mask_split_list[index_even])
+        concat_data(mask_even, dim=2).save(file_mask_even, verbose=0)  # Concatenate in z-axis and save
+
+        mask_odd = []
+        # loop across odd slices
+        for index_odd in range(1, len(im_mask_split_list), 2):
+            mask_odd.append(im_mask_split_list[index_odd])
+        concat_data(mask_odd, dim=2).save(file_mask_odd, verbose=0)  # Concatenate in z-axis and save
+
+    # Run moco only on even slices
+    sct.printv('\nStarting moco on even slices...', param.verbose)
+    param.fname_data = 'data_even.nii'
+    if param.fname_mask != '':
+        param.fname_mask = 'mask_even.nii'
+    im_data_even = moco_wrapper(param)
+
+    # Run moco only on odd slices
+    sct.printv('\nStarting moco on odd slices...', param.verbose)
+    param.fname_data = 'data_odd.nii'
+    if param.fname_mask != '':
+        param.fname_mask = 'mask_odd.nii'
+    im_data_odd = moco_wrapper(param)
+
+    # TODO: files moco_params_x, moco_params_y.nii.gz and moco_params.tsv are overwritten now (even by odd)
+    #       - Do we want to preserve these files for both subdatasets or merge them?
```

Comment:
ideally we would like to save them into two separate folders (with explicit suffix _odd and _even)